<launch>
  <arg name="robot_namespace" default="robot"/>

  <param name="use_sim_time" value="true"/>

  <!-- Artifact localization -->
  <include file="$(find artifact_localization)/launch/artifact_localization.launch">
    <arg name="robot_namespace" value="$(arg robot_namespace)"/>
  </include>
  <rosparam ns="$(arg robot_namespace)">
    artifact_localization:
      # Remap image topics
      rgb:
        image_topic: color/image_raw_throttle
      depth:
        image_topic: aligned_depth_to_color/image_raw_throttle
  </rosparam>

  <!-- TF -->
  <include file="$(find capability_common)/launch/robot_state_publisher.launch">
    <arg name="robot_namespace" value="$(arg robot_namespace)"/>
  </include>

  <!-- Image decomporessor -->
  <include file="$(find sensor_image_utils)/launch/rs_compressed_to_raw.launch">
    <arg name="robot_namespace" value="$(arg robot_namespace)"/>
    <arg name="camera_name" value="camera_front"/>
    <arg name="throttle" value="true"/>
    <arg name="image_in" value="image_raw"/>
    <arg name="image_out" value="image_raw"/>
  </include>

  <include file="$(find sensor_image_utils)/launch/rs_compressed_to_raw.launch">
    <arg name="robot_namespace" value="$(arg robot_namespace)"/>
    <arg name="camera_name" value="camera_left"/>
    <arg name="throttle" value="true"/>
    <arg name="image_in" value="image_raw"/>
    <arg name="image_out" value="image_raw"/>
  </include>

  <include file="$(find sensor_image_utils)/launch/rs_compressed_to_raw.launch">
    <arg name="robot_namespace" value="$(arg robot_namespace)"/>
    <arg name="camera_name" value="camera_right"/>
    <arg name="throttle" value="true"/>
    <arg name="image_in" value="image_raw"/>
    <arg name="image_out" value="image_raw"/>
  </include>

  <!-- Visualization -->
  <node pkg="rviz"
        type="rviz"
        name="rviz"
        args="-d $(dirname)/rerun_artifact_$(arg robot_namespace).rviz"
        ns="$(arg robot_namespace)"/>

</launch>
